### PR TITLE
x86-16 few fixes

### PR DESCRIPTION
**Pull request by lasek0**
**Checklist**

- [x] Closing issues: #issue
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [radare2book](https://github.com/radareorg/radare2book)

**Description**

all related only to x86-16
- fix displaay of seg:off addresses in call (https://github.com/radareorg/radare2/issues/18888 - CLOSED now)
- fix seg:off analysis allowing to replace addr with func name
- fix jmptbl analysis for some addressess, it was incorrectly rejected by validate function
- ~~preserve segoff parameter while printing disassembly~~ (#19599)
- fix analysis of instructions argument for lds and les (like lea) (https://github.com/radareorg/radare2/issues/18890 - CLOSED now)

sorry for the additional commits in that pull-request i screw up my github repo, hope it will be not a problem.

### CODE REVIEW

**Comment by lasek0** [comment]
looks like I rebased my repo with failing master. I will wait until master become green.

File: libr/anal/jmptbl.c
Comment (Lazula) [inline libr/anal/jmptbl.c:239]: align closing bracket

File: libr/anal/jmptbl.c
Comment (Lazula) [inline libr/anal/jmptbl.c:195]: wrong indentation, and use `strncmp()` like `is_arm` above and check `is_arm` before calling `strncmp()`.

```suggestion
    bool is_x86 = is_arm? false: !strncmp (anal->cur->arch, "x86", 3);
```

File: libr/anal/p/anal_x86_cs.c
Comment (Lazula) [inline libr/anal/p/anal_x86_cs.c:1071]: spacing around parens `if () {`

File: libr/anal/p/anal_x86_cs.c
Comment (Lazula) [inline libr/anal/p/anal_x86_cs.c:1080]: spacing around parens

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:153]: use `#if` for commenting out code

```suggestion
#if 1
int segoff = 0;
#else
// TODO: respect segoff
r_config_get_i (p->analb.anal->coreb.core, "asm.segoff");
#endif
```

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:196]: spacing around parens

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:202]: there should be spaces around else: `} else {`

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:416]: space between operators and operands

```suggestion
                        if (x86 && bits == 16 && *pnum == ':') {
```

**Review by Lazula — COMMENTED** [review]
some style changes

**Comment by Lazula** [comment]
> looks like I rebased my repo with failing master. I will wait until master become green.

don't worry about merging on failing master, its fine

>  sorry for the additional commits in that pull-request i screw up my github repo, hope it will be not a problem.

we can squash into 1 commit when merging if they are messy, not a problem :)

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:195]: `strstr (`

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c:200]: space around operator `(s << seggrn)`

**Review by Lazula — COMMENTED** [review]
noticed 2 more

**Comment by lasek0** [comment]
Yea thanks, should be fine now. I will be more patient to syntax in future.

**Comment by trufae** [comment]
04E88C87-315E-459A-807C-D011BB73608F

File: libr/parse/filter.c
Comment (Lazula) [inline libr/parse/filter.c]: if you arent using it but still want the comment you can do so, the complaint happened because of the `#if 1` part.

```suggestion
#if 0
// TODO: respect segoff
int segoff = r_config_get_i (p->analb.anal->coreb.core, "asm.segoff");
#endif
```

**Comment by Lazula** [comment]
you can include that commented bit if you want, otherwise no further input from me :+1:

**Review by Lazula — APPROVED** [review]
style approval, no functionality opinion. will leave that to @trufae

File: libr/core/cmd_print.c
Comment (trufae) [inline libr/core/cmd_print.c:5359]:
```suggestion
		bool segoff = r_config_get_b (core->config, "asm.segoff");
```

File: libr/core/cmd_print.c
Comment (trufae) [inline libr/core/cmd_print.c:5839]:
```suggestion
		r_config_set_b (core->config, "asm.segoff", segoff);
```

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:151]:
```suggestion
	const int bits = p->analb.anal->bits;
```

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:152]:
```suggestion
	const int seggrn = p->analb.anal->seggrn;
```

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:198]: Use rnumget instead

Comment (trufae) [reply libr/parse/filter.c:198]:
```suggestion
			ut64 o = r_num_get (NULL, colon + 1);
```

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:197]:
```suggestion
			ut64 s = r_num_get (NULL, ptr);
```

File: libr/anal/p/anal_x86_cs.c
Comment (trufae) [inline libr/anal/p/anal_x86_cs.c:1079]: Unnecessary braces

File: libr/anal/p/anal_x86_cs.c
Comment (trufae) [inline libr/anal/p/anal_x86_cs.c:1070]: Unnecessary braces

**Review by trufae — CHANGES_REQUESTED** [review]
Please add a test

File: libr/core/cmd_print.c
Comment (lasek0) [reply libr/core/cmd_print.c]: unfortunatelly I need to revert that part because it is more complicated problem than I though at first place. #19599

File: libr/core/cmd_print.c
Comment (lasek0) [reply libr/core/cmd_print.c]: unfortunately I need to revert that part because it is more complicated problem than I though at first place. #19599

**Comment by lasek0** [comment]
actually during preparing tests I found that preserving segoff in x86-16 bit is not fixed in that proposed pull-request, I file an separate issue #19599 for that, and I will try to fix that separately. If You can point me (even Your first though without deep checking) what check first I will be happy because debugging time shorts a lot.

**Comment by lasek0** [comment]
is something changed on master in x86-16bit subject?
```
$ r2 -v
radare2 5.5.5 27258 @ linux-x86-64 git.5.5.4
commit: 7db2a30b56e3cb7a956cd4b9e61317b8f3b3fcf7 build: 2022-01-20__01:14:11
```
test:
```
e asm.arch=x86
e asm.bits=16
wx 9090558bec9a0d0000005dc39090558bec5dc3
pd 10
```

```
[0000:0000]> pd 10
...
            0000:0003      8bec           mov bp, sp
            0000:0005      9a0d000000     lcall 0, 0xd
            0000:000a      5d             pop bp
...
```

now seg:off is not displayed at all. previously it looks like this: 
            0000:0005      9a0d000000     lcall 0:0xd
but now is:
            0000:0005      9a0d000000     lcall 0, 0xd

I have troubles during tests because of that, this change cause my fixes to be not compatible any more. looks like disassembly is changed. or do I mess-up my environment?

File: libr/anal/jmptbl.c
Comment (trufae) [inline libr/anal/jmptbl.c:195]:
```suggestion
	bool is_x86 = !is_arm && anal->cur->arch && !strncmp (anal->cur->arch, "x86", 3);
```

File: libr/anal/jmptbl.c
Comment (trufae) [inline libr/anal/jmptbl.c:196]:
```suggestion
	const bool is_v850 = !is_arm && !is_x86 && ((anal->cur->arch && !strncmp (anal->cur->arch, "v850", 4)) || !strncmp (anal->coreb.cfgGet (anal->coreb.core, "asm.cpu"), "v850", 4));
```

**Review by trufae — CHANGES_REQUESTED** [review]
Looks good! Thanks for the test its clearer now. Please consider applying my patch suggestions.

File: libr/core/cmd_print.c
Comment (trufae) [reply libr/core/cmd_print.c]: Ok

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:198]:
```suggestion
			ut64 o = r_num_get (NULL, colon + 1);
```

Comment (trufae) [inline libr/parse/filter.c]:
```suggestion
			ut64 o = r_num_get (NULL, colon + 1);
```

File: libr/parse/filter.c
Comment (trufae) [inline libr/parse/filter.c:202]:
```suggestion
			off = r_num_get (NULL, ptr);
```

**Comment by trufae** [comment]
Just applied my suggestions. Waiting for the CI and i'll merge

**Comment by trufae** [comment]
Merged by hand into master! thank you!

Please try not to use merge commits in PRs, force push makes it cleaner to review and maintain. i just squashed the commits grouping them by topics before pushing to master

**Pull request closed without merging**
